modules/xmpp/RtxModifier.js

```diff
@@ -163,62 +110,64 @@ export default class RtxModifier {
      * @param {string} sdpStr sdp in raw string format
      */
     modifyRtxSsrcs (sdpStr) {
-        let parsedSdp = transform.parse(sdpStr);
-        let videoMLine = 
-            parsedSdp.media.find(mLine => mLine.type === "video");
-        if (videoMLine.direction === "inactive" ||
-                videoMLine.direction === "recvonly") {
+        const sdpTransformer = new SdpTransformWrap(sdpStr);
+        if (!sdpTransformer.selectMedia("video")) {
+            logger.error("No 'video' media found in the sdp: " + sdpStr);
+            return sdpStr;
+        }
+        const direction = sdpTransformer.mediaDirection;
+        if (direction === "inactive" || direction === "recvonly") {
             logger.debug("RtxModifier doing nothing, video " +
                 "m line is inactive or recvonly");
             return sdpStr;
         }
-        if (!videoMLine.ssrcs) {
+        if (sdpTransformer.getSSRCCount() < 1) {
           logger.debug("RtxModifier doing nothing, no video ssrcs present");
           return sdpStr;
         }
         logger.debug("Current ssrc mapping: ", this.correspondingRtxSsrcs);
-        let primaryVideoSsrcs = getPrimaryVideoSsrcs(videoMLine);
+        let primaryVideoSsrcs = sdpTransformer.getPrimaryVideoSSRCs();
         logger.debug("Parsed primary video ssrcs ", primaryVideoSsrcs, " " +
             "making sure all have rtx streams");
-        primaryVideoSsrcs.forEach(ssrc => {
-            let msid = SDPUtil.getSsrcAttribute(videoMLine, ssrc, "msid");
-            let cname = SDPUtil.getSsrcAttribute(videoMLine, ssrc, "cname");
+        for (const ssrc of primaryVideoSsrcs) {
+            let msid = sdpTransformer.getSSRCAttrValue(ssrc, "msid");
+            let cname = sdpTransformer.getSSRCAttrValue(ssrc, "cname");
             let correspondingRtxSsrc = this.correspondingRtxSsrcs.get(ssrc);
             if (correspondingRtxSsrc) {
                 logger.debug("Already have an associated rtx ssrc for " +
-                    " video ssrc " + ssrc + ": " + 
+                    " video ssrc " + ssrc + ": " +
```

Comment:
this is all (my) existing code, so not pawel's fault :) guessing this will cause too many conflicts for him to fix now